File: EasyMarkets/app/src/main/java/merryweather/com/easymarkets/di/component/ActivityComponent.java
package merryweather.com.easymarkets.di.component;


import android.content.Context;

import dagger.Component;
import merryweather.com.easymarkets.MainActivity;
import merryweather.com.easymarkets.di.modules.ActivityModule;
import merryweather.com.easymarkets.di.scopes.PerActivity;
import merryweather.com.easymarkets.repository.AppartmentRepository;


/**
 * Created by S on 17.05.2018.
 */

@PerActivity
@Component(dependencies = AppComponent.class, modules = ActivityModule.class)
public interface ActivityComponent {

    Context context();

    AppartmentRepository appartmentRepository();

    void inject(MainActivity activity);


}

File: Nista/app/src/main/java/com/example/nista/di/component/ActivityComponent.kt
package com.example.nista.di.component


import android.content.Context
import com.example.nista.list.MovieListActivity
import com.example.nista.api.MovieService
import com.example.nista.detail.MovieActivity
import com.example.nista.repository.MovieRepository
import com.leroymerlin.cds.di.modules.ActivityModule
import com.leroymerlin.cds.di.modules.ViewModelModule
import com.leroymerlin.cds.di.scopes.PerActivity
import dagger.Component


@PerActivity
@Component(dependencies = arrayOf(AppComponent::class), modules = arrayOf(ActivityModule::class, ViewModelModule::class))
interface ActivityComponent {

    fun context(): Context

    fun movieService(): MovieService

    fun movieRepository(): MovieRepository


    fun inject(activity: MovieListActivity)

    fun inject(activity: MovieActivity)

}
